src/problems/basic_problems.jl

```diff
@@ -688,6 +688,14 @@ function OptimizationProblem(f, args...; kwargs...)
     OptimizationProblem{true}(OptimizationFunction{true}(f), args...; kwargs...)
 end
 
+function OptimizationProblem(prob::NonlinearLeastSquaresProblem; kwargs...)
+    if isinplace(prob)
+        throw(ArgumentError("Converting NonlinearLeastSquaresProblem to OptimizationProblem is not supported with in-place functions yet."))
+    end
+    optf = OptimizationFunction(sum ∘ prob.f, grad = (Jv, u, p) -> prob.f.jvp(Jv, prob.f(u, p), u, p), kwargs...)
```

Comment:
`sum(abs2, prob.f(u,p))`, not just `sum`

Reply:
I think the loss for NLLS already has the square

Reply:
@avik-pal wait, I thought the interface would be the same as nonlinearsolve, i.e. returning the vector of residuals. `f(du,u,p)` and then `sum(abs2,du)` is the loss. Now I'm looking at the tests and see the user is required to write it. Having that summation be implicit enforces it's actually a least squares problem.

Reply:
Right it is same as NonlinearProblem, `sum ∘ prob.f` should be `sum(abs2,....)`

Reply:
IIUC what Chris meant was the other way around, the NLLS problem should do the squaring implicitly and not ask the user to do that (and then the loss here would just be the sum)

Reply:
Okay I see what happened. https://github.com/SciML/NonlinearSolve.jl/commit/b8aca894e5168085f7f06ff656506c107f8788c6 the original test was just weird, but it was since fixed https://github.com/SciML/NonlinearSolve.jl/commit/883a3928cb4bf6893ea9a18e2c362b520891d5af. This means it's `sum(abs2,...)`.

Reply:
Ah yes, I had seen the previous version and added this